modules/dmdIdSystem.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * This module adds dmdId to the User ID module
+ * The {@link module:modules/userId} module is required
+ * @module modules/dmdIdSystem
+ * @requires module:modules/userId
+ */
+
+import * as utils from '../src/utils.js';
+import { submodule } from '../src/hook.js';
+
+/** @type {Submodule} */
+export const dmdIdSubmodule = {
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: 'dmdId',
+
+  /**
+   * decode the stored id value for passing to bid requests
+   * @function decode
+   * @param {(Object|string)} value
+   * @returns {(Object|undefined)}
+   */
+  decode(value) {
+    return value && typeof value === 'string'
+      ? { 'dmdId': value }
+      : undefined;
+  },
+
+  /**
+   * performs action to obtain id and return a value in the callback's response argument
+   * @function getId
+   * @param {SubmoduleConfig} [config]
+   * @param {ConsentData}
+   * @param {Object} cacheIdObj - existing id, if any consentData]
+   * @returns {IdResponse|undefined}
+   */
+  getId(config, consentData, cacheIdObj) {
+    try {
+      const configParams = (config && config.params) || {};
+      if (
+        !configParams ||
+        !configParams.api_key ||
+        typeof configParams.api_key !== 'string'
```

Comment:
what does the `api_key` do, considering you don't check it's validity or anything? meaning, all i have to do is add any string and this will allow it to pass through

Reply:
@smenzer In a future release, DMD plans to introduce reading the authentication ID against a REST API. The REST API will validate the API Key on the backend. Our customers typically are unwilling to change integrations after the initial installation. We would like to standardize our configuration to require an API Key in this release so our customers do not need to revise their integration after we introduce the REST API in a subsequent release.

Reply:
thanks for the clarification. why not build the rest api now even if you're not authenticating so that publishers won't have to upgrade prebid to get the update (which, as you mentioned, they don't like having to make changes)?

Reply:
@smenzer DMD is working on the next release now, but we wanted to submit the 1st party authentication ID feature first as this feature addition is low in complexity and thought it would be better for us to learn how to contribute features to Prebid with a low complexity change. If you would prefer us to remove the expectation of the api_key in the configuration for this release, we can remove it.

Reply:
I don't have a problem with the key being in there, I just wanted to point out that by not building to your rest api now, you'll still be requiring publishers to make a change (upgrade prebid) when it's ready. if you're comfortable with that, i have no objections to leaving the config as you have it.

Reply:
@smenzer Thank you. We appreciate the understanding. You are correct and we will note to remind publishers that a prebid rebuild is required when we release the REST API addition.